### ISSUE TITLE
[49.8 Testing Branch] Uncaught Concurrent Modification Exception when launching MM

### ISSUE THREAD

**Issue by Thom293** [MegaMek/megamek#3617]
21:22:13,419 INFO  [megamek.MegaMek] {main}
megamek.MegaMek.initializeLogging(MegaMek.java:115) - Starting MegaMek v0.49.8-SNAPSHOT
	Build Date: 2022-05-03T01:13:29.469020
	Release Date: 2022-02-18
	Today: 2022-05-03
	Origin Project: MekHQ
	Java Vendor: AdoptOpenJDK
	Java Version: 11.0.10
	Platform: Windows 10 10.0 (amd64)
	System Locale: en_US
	Total memory available to MegaMek: 2 GB

Launching a Capture mission.  Happened in MM so I posted here.  Might be a MHQ issue though.

I did reinforce this battle with a lance already deployed on the same stratcon track.  First time I have ever tried to do that, if it matters.
Closed and reloaded and reinforced with a different undeployed lance.  Same result.
Closed and reloaded with no reinforcement and it launched normally.

[megamek.log](https://github.com/MegaMek/megamek/files/8617649/megamek.log)
[mekhq.log](https://github.com/MegaMek/megamek/files/8617650/mekhq.log)
[The Bluecoats30621112.cpnx.gz](https://github.com/MegaMek/megamek/files/8617661/The.Bluecoats30621112.cpnx.gz)
![Screenshot (426)](https://user-images.githubusercontent.com/23645569/166618239-74a57429-667e-4053-a685-6be6d8c2be43.png)

**Comment by Thom293** [comment]
Blarg.  Completed scenario and got another one when it asked me if I wanted to save all units.  I dismissed it and it DID allow me to save the mul.  Made a manual MM save too. Had to close MM with the X.  Didnt give me the scenario resolution dialogue, but I was able to resolve it manually with mul.  Screen and ending files attached.

![Screenshot 2022-05-03 223430](https://user-images.githubusercontent.com/23645569/166620316-6e6f251a-94f1-4bc7-b2b7-84b3d65de764.png)
[mm.sav.gz](https://github.com/MegaMek/megamek/files/8617762/mm.sav.gz)
[The Bluecoats30621112B.cpnx.gz](https://github.com/MegaMek/megamek/files/8617763/The.Bluecoats30621112B.cpnx.gz)
[megamek.log](https://github.com/MegaMek/megamek/files/8617764/megamek.log)
[mekhq.log](https://github.com/MegaMek/megamek/files/8617765/mekhq.log)

**Comment by Windchild292** [comment]
Here is correct... I've just not figured out how to get that message to change based on launching program without some serious internal changes.

**Comment by Windchild292** [comment]
```
22:33:46,102 ERROR [mekhq.MekHQ] {Packet Pump}
mekhq.MekHQ.lambda$main$0(MekHQ.java:234) - Uncaught Exception Detected
java.util.ConcurrentModificationException
	at java.base/java.util.Vector$VectorSpliterator.forEachRemaining(Vector.java:1494)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at megamek.server.Server.send(Server.java:30879)
	at megamek.server.Server.prepareForPhase(Server.java:2545)
	at megamek.server.Server.changePhase(Server.java:2236)
	at megamek.server.Server.endCurrentPhase(Server.java:2974)
	at megamek.server.Server.checkReady(Server.java:2022)
	at megamek.server.Server.handle(Server.java:31013)
	at megamek.server.Server$PacketPump.run(Server.java:173)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by Windchild292** [comment]
This is resolved in #3602, and is part of the new testing branches.
